paymentsheet/src/main/java/com/stripe/android/paymentsheet/state/PaymentSheetLoader.kt

```diff
@@ -347,3 +350,22 @@ private fun List<PaymentMethod>.withLastUsedPaymentMethodFirst(
 private fun PaymentMethod.toPaymentSelection(): PaymentSelection.Saved {
     return PaymentSelection.Saved(this, isGooglePay = false)
 }
+
+private fun PaymentSheet.BillingDetailsCollectionConfiguration.toInternal(): BillingDetailsCollectionConfiguration {
+    return BillingDetailsCollectionConfiguration(
+        collectName = name == Always,
+        collectEmail = email == Always,
+        collectPhone = phone == Always,
+        address = when (address) {
+            PaymentSheet.BillingDetailsCollectionConfiguration.AddressCollectionMode.Automatic -> {
+                BillingDetailsCollectionConfiguration.AddressCollectionMode.Automatic
+            }
+            PaymentSheet.BillingDetailsCollectionConfiguration.AddressCollectionMode.Never -> {
+                BillingDetailsCollectionConfiguration.AddressCollectionMode.Never
+            }
+            PaymentSheet.BillingDetailsCollectionConfiguration.AddressCollectionMode.Full -> {
+                BillingDetailsCollectionConfiguration.AddressCollectionMode.Full
+            }
+        },
+    )
```

Comment:
This conversion doesn't work because it depends on payment method, i.e. for Cards name and email will only be collected if they are "Always", but for bank account they also show for "Automatic"

Reply:
Chatted offline. The conversion should be fine, as US Bank Account has it’s own separate handling.